tests/inference_on_device_test.py

```diff
@@ -311,142 +281,21 @@ def test_train_with_different_data_and_training_device(
     x_o = torch.zeros(x.shape[1])
     inference = inference.append_simulations(theta, x, data_device=data_device)
 
-    posterior_estimator = inference.train(max_num_epochs=2)
+    estimator = inference.train(max_num_epochs=2)
 
     # Check for default device for inference object
     weights_device = next(inference._neural_net.parameters()).device
     assert torch.device(training_device) == weights_device
 
-    _ = DirectPosterior(
-        posterior_estimator=posterior_estimator, prior=prior
+    # Check device inference in posterior class
+    posterior = inference.build_posterior(
+        # use data_device as switch to test both device inference cases.
+        density_estimator=estimator if data_device == "cpu" else None,
+        prior=prior,
     ).set_default_x(x_o)
-
-
-@pytest.mark.gpu
-@pytest.mark.parametrize(
-    "inference_method", [SNPE_A, SNPE_C, SNRE_A, SNRE_B, SNRE_C, SNLE]
-)
-@pytest.mark.parametrize("prior_device", ("cpu", "gpu"))
-@pytest.mark.parametrize("embedding_net_device", ("cpu", "gpu"))
-@pytest.mark.parametrize("data_device", ("cpu", "gpu"))
-@pytest.mark.parametrize("training_device", ("cpu", "gpu"))
-def test_embedding_nets_integration_training_device(
```

Comment:
why did you remove this test?

Reply:
The test is obsolete now because we assume that `embedding_net_device == training_device` and there is another test for the `training_device` `data_device` options.  
But we could change it to test the case that a custom NN (with embedding) has different device than the passed `training_device`. I will do that.